api/envoy/api/v2/core/base.proto

```diff
@@ -79,6 +79,13 @@ message Node {
   // version of Envoy is being tested in a heterogeneous fleet. This will be set
   // by Envoy in management server RPCs.
   string build_version = 5;
+
+  // A list of client capabilities indicating extensions that can be provided.
+  // Strings must be in reverse DNS notation. The following values are known:
+  //
+  // "envoy.grpc_client": Indicates that the client is a gRPC client and not a proxy. For
+  //     LDS, the server is required to provide exactly one HTTP API listener.
+  repeated string capabilities = 6;
```

Comment:
Consider adding the full set of features in https://docs.google.com/document/d/1xeVvJ6KjFBkNjVspPbY_PwEDHC7XPi0J5p1SqUXcCl8/edit#heading=h.7mzjb4t3khsf. @yanavlasov might be able to assist on the Envoy side implementation here as part of his work contributing to the v3 API (Yan; I reckon this would be a very helpful initial step, implementing things like user agent, filter reporting etc.).

Reply:
Per above I would suggest we first agree on what we want built-in versus opaque.

Additionally, it minimum, I would future proof this by making a `Capability` wrapper message as I could easily see capabilities eventually having more info than just a string name.

Reply:
Thanks, Harvey; I didn't realize that the versioning doc actually spelled this out.  I'll update this once we are converging on some of the details.

Reply:
I've removed the capability proposal from this PR, since I think we no longer need it for this functionality.